app/code/Magento/InventoryExportStock/Model/ResourceModel/StockIndexDumpProcessor.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryExportStock\Model\ResourceModel;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Select;
+use Magento\InventoryExportStock\Model\GetQtyForNotManageStock;
+use Magento\InventoryIndexer\Model\StockIndexTableNameResolverInterface;
+use Magento\InventorySales\Model\ResourceModel\IsStockItemSalableCondition\ManageStockCondition;
+use Zend_Db_Expr;
+use Zend_Db_Select_Exception;
+
+/**
+ * Class GetStockIndexDump provides sku and qty of products dumping them from stock index table
+ */
+class StockIndexDumpProcessor
+{
+    /**
+     * @var StockIndexTableNameResolverInterface
+     */
+    private $stockIndexTableNameResolver;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var ManageStockCondition
+     */
+    private $manageStockCondition;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $connection;
+
+    /**
+     * @var GetQtyForNotManageStock
+     */
+    private $getQtyForNotManageStock;
+
+    /**
+     * GetStockIndexDump constructor
+     *
+     * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+     * @param ResourceConnection $resourceConnection
+     * @param ManageStockCondition $manageStockCondition
+     * @param GetQtyForNotManageStock $getQtyForNotManageStock
+     */
+    public function __construct(
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        ResourceConnection $resourceConnection,
+        ManageStockCondition $manageStockCondition,
+        GetQtyForNotManageStock $getQtyForNotManageStock
+    ) {
+        $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+        $this->resourceConnection = $resourceConnection;
+        $this->manageStockCondition = $manageStockCondition;
+        $this->getQtyForNotManageStock = $getQtyForNotManageStock;
+    }
+
+    /**
+     * Provides sku and qty of products dumping them from stock index table
+     *
+     * @param int $stockId
+     * @return array
+     * @throws Zend_Db_Select_Exception
```

Comment:
this is high-level service , don't return `Zend_Db_Select_Exception` out here

catch (Exception $e) and re-throw more high-level exception